src/thor/triplegbuilder.cc

```diff
@@ -943,12 +943,9 @@ void AccumulateRecostingInfoForward(const valhalla::Options& options,
   }
 
   // setup a callback for the recosting to get each edge
-  auto in_itr = leg.node().begin();
-  sif::EdgeCallback edge_cb = [&in_itr]() -> baldr::GraphId {
-    auto edge_id = in_itr->has_edge() ? baldr::GraphId(in_itr->edge().id()) : baldr::GraphId{};
-    ++in_itr;
-    return edge_id;
-  };
+  std::vector<GraphId> edge_ids;
+  for (auto in_itr = leg.node().begin(); in_itr->has_edge(); ++in_itr)
+    edge_ids.emplace_back(in_itr->edge().id());
```

Comment:
if switching from the callback model didnt help performance i would suggest we revert that change. do you konw if that change made a difference in performance?

Reply:
will check